File: server/DocumentsKM/Models/MarkOperatingConditions.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DocumentsKM.Models
{
    // Общие условия эксплуатации
    public class MarkOperatingConditions
    {
        // Марка
        [ForeignKey("MarkId")]
        public virtual Mark Mark { get; set; }
        [Key]
        public Int32 MarkId { get; set; }

        // Коэффициент надежности по ответственности
        [Required]
        public float SafetyCoeff { get; set; }

        // Агрессивность среды
        [Required]
        [ForeignKey("EnvAggressivenessId")]
        public virtual EnvAggressiveness EnvAggressiveness { get; set; }

        // Температура
        [Required]
        public Int16 Temperature { get; set; }

        // Зона эксплуатации
        [Required]
        [ForeignKey("OperatingAreaId")]
        public virtual OperatingArea OperatingArea { get; set; }

        // Группа газов
        [Required]
        [ForeignKey("GasGroupId")]
        public virtual GasGroup GasGroup { get; set; }

        // Материал конструкций
        [Required]
        [ForeignKey("ConstructionMaterialId")]
        public virtual ConstructionMaterial ConstructionMaterial { get; set; }

        // Тип лакокрасочного материала
        [Required]
        [ForeignKey("PaintworkTypeId")]
        public virtual PaintworkType PaintworkType { get; set; }

        // Высокопрочные болты
        [Required]
        [ForeignKey("HighTensileBoltsTypeId")]
        public virtual HighTensileBoltsType HighTensileBoltsType { get; set; }
    }
}

File: client/src/model/MarkOperatingConditions.ts
import EnvAggressiveness from './EnvAggressiveness'
import OperatingArea from './OperatingArea'
import GasGroup from './GasGroup'
import ConstructionMaterial from './ConstructionMaterial'
import PaintworkType from './PaintworkType'
import HighTensileBoltsType from './HighTensileBoltsType'

interface IMarkOperatingConditions {
    safetyCoeff: number
    envAggressiveness: EnvAggressiveness
    temperature: number
    operatingArea: OperatingArea
    gasGroup: GasGroup
    constructionMaterial: ConstructionMaterial
    paintworkType: PaintworkType
    highTensileBoltsType: HighTensileBoltsType
}

class MarkOperatingConditions {
    safetyCoeff: number
    envAggressiveness: EnvAggressiveness
    temperature: number
    operatingArea: OperatingArea
    gasGroup: GasGroup
    constructionMaterial: ConstructionMaterial
    paintworkType: PaintworkType
    highTensileBoltsType: HighTensileBoltsType

	constructor(obj?: IMarkOperatingConditions) {
        this.safetyCoeff = (obj && obj.safetyCoeff) || 0
        this.envAggressiveness = (obj && obj.envAggressiveness) || null
        this.temperature = (obj && obj.temperature) || 0
        this.operatingArea = (obj && obj.operatingArea) || null
        this.gasGroup = (obj && obj.gasGroup) || null
        this.constructionMaterial = (obj && obj.constructionMaterial) || null
        this.paintworkType = (obj && obj.paintworkType) || null
        this.highTensileBoltsType = (obj && obj.highTensileBoltsType) || null
    }
}

export default MarkOperatingConditions
